daft/io/catalog.py

```diff
@@ -0,0 +1,32 @@
+# isort: dont-add-import: from __future__ import annotations
+
+from dataclasses import dataclass
+from enum import Enum
+from typing import Optional
+
+
+class DataCatalog(Enum):
+    """Supported data catalogs."""
+
+    GLUE = "glue"
+    """
+    AWS Glue Data Catalog.
+    """
+    UNITY = "unity"
+    """
+    Databricks Unity Catalog.
+    """
+
+
+@dataclass
+class DataCatalogTable:
```

Comment:
I'm hoping that we can find a way to implement catalog discovery + format discovery for Glue so the user doesn't need to specify it in the `daft.read_catalog_table()` API, whose form would then be something like
```python
df = daft.read_catalog_table("catalog_name.database_name.table_name")
```
To start, we could ask the user to set `default_catalog="glue"` in `DaftContext`, since it's not likely that they'd be using multiple catalogs at once.

I've looked into automatic format discovery (Iceberg, Delta Lake, etc.) for Glue, but haven't figured out how they do it yet. Planning to save all of this for a future PR.